### ISSUE TITLE
phpList - CKEditor images not working (and other improvements)

### ISSUE THREAD

**Issue by JedMeister** [turnkeylinux/tracker#1278]
Matthew reported in the [forums](https://www.turnkeylinux.org/comment/30953#comment-30953) that he was having troubles with the images component of CKEditor in our phpList appliance. The error that he was getting was: `HTTP error occurred during file upload (404: File not found)`.

Following his instructions I was able to recreate the issue. It turns out that there are a few things that we should tweak with the phpList appliance (detailed below), but the main one causing the 404 is the `Path to KCFinder` setting in the ckeditor settings. By default it is set to `/var/www/phplist/admin/plugins/CKEditorPlugin/kcfinder` (the absolute file path). However, it should be relative to the admin web path. I.e. `plugins/CKEditorPlugin/kcfinder`.

----

Workaround
----------------

This can be adjusted via the settings: Admin >> Config >> Manage plugins >> CKEditor Settings (within the CKEditorPlugin area). Then change the "Path to KCFinder" to `plugins/CKEditorPlugin/kcfinder`.

Also the php-gd extension should be installed to support the display of images:
```
apt update
apt install php-gd
service apache2 restart
```

----

Roundup of issues that should be fixed for the next release of phpList:

- [x] set `'kcfinder_path','plugins/CKEditorPlugin/kcfinder'` (in DB - `phplist.config`)
- [x] inclusion of `php-gd` in the plan
- [x] inclusion of `php-zip` in the plan (support for downloaded plugins)
- [x] chown the custom plugin dir to `www-data` (avoid warning)
- [x] create symlinks to the default included plugins so that they work OOTB. I.e.:
```
chown -R www-data:www-data /var/www/phplist-plugins
for path in /var/www/phplist/admin/plugins/*; do
    ln -s $path /var/www/phplist-plugins/$(basename $path)
done
```
- [x] disable Apache indexes (i.e. `-Indexes` in the Apache config: `/etc/apache2/sites-avaialble/phplist.conf`)

**Comment by JedMeister** [comment]
Closed by https://github.com/turnkeylinux-apps/phplist/commit/3260e0dcce16e336acf0932d001526f3580234b4

**Comment by JedMeister** [comment]
Yes that is correct. Apologies that that wasn't explicitly mentioned. Perhaps I should note it somewhere on the readme (or similar)?!

We don't pre-create that as as per upstream (i.e. the phpList developers) note, it does open a potential security hole. The risk is probably low, but it's best to be aware of that when you enable it. The rationale is that if it's enabled by default, users aren't aware of the potential risks.

**Comment by miningandhydraulics** [comment]
> 
> 
> Yes that is correct. Apologies that that wasn't explicitly mentioned. Perhaps I should note it somewhere on the readme (or similar)?!
> 
> We don't pre-create that as as per upstream (i.e. the phpList developers) note, it does open a potential security hole. The risk is probably low, but it's best to be aware of that when you enable it. The rationale is that if it's enabled by default, users aren't aware of the potential risks.

That would be a help adding it into the readme :) and maybe adding the new version into your image

**Comment by JedMeister** [comment]
Firstly, I'm not 100% sure what I was thinking when I did my last post above. It must have been a long day... :smile: 

Anyway, I will be rebuilding the phpList appliance shortly and we'll publish an updated version with all the improvements I've made. Unfortuantely, it may not be available until early in the new year as I'm off for a couple of weeks tomorrow.

But thanks again for bringing the issue(s) to my attention! :+1:
